### ISSUE TITLE
HTTPS default port should be 443 and not 80

### ISSUE THREAD

**Issue by PFadel** [Kong/kong#3441]
### Summary

Hello everyone.

When I register a service using the `url` parameter with a https address and the port value is not in the url string ( something like: https://www.google.com.br/ ), Kong register the service with port 80. This causes the following error:

```
[error] 13997#0: *815304 SSL_do_handshake() failed (SSL: error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol) while SSL handshaking to upstream
```

When I use the PATCH service route to set the port to 443, everything works fine.

### Steps To Reproduce

1. Add a Google service with url set to: "https://www.google.com.br/".
2. Add route to the service.
3. Execute a request, see the error.
4. PATCH service to port 443.
5. Execute another request and see that it works.

Since Kong is able to understand the protocol in the url parameter, I believe it should set the default port accordingly, 80 to http services and 443 to https services.

I know it's a small matter but I though it shouldn't pass unnoticed. :smiley: 

### Additional Details & Logs

- Kong version: 0.13.0
- Kong debug-level startup logs:
`2018/05/08 16:35:20 [verbose] Kong: 0.13.0
2018/05/08 16:35:20 [debug] ngx_lua: 10011
2018/05/08 16:35:20 [debug] nginx: 1013006
2018/05/08 16:35:20 [debug] Lua: LuaJIT 2.1.0-beta3
2018/05/08 16:35:20 [info] Kong stopped
2018/05/08 16:35:20 [verbose] reading config file at /etc/kong/kong.conf
2018/05/08 16:35:20 [debug] admin_access_log = "logs/admin_access.log"
2018/05/08 16:35:20 [debug] admin_error_log = "logs/error.log"
2018/05/08 16:35:20 [debug] admin_listen = {"127.0.0.1:8001","127.0.0.1:8444 ssl"}
2018/05/08 16:35:20 [debug] anonymous_reports = true
2018/05/08 16:35:20 [debug] cassandra_consistency = "ONE"
2018/05/08 16:35:20 [debug] cassandra_contact_points = {"127.0.0.1"}
2018/05/08 16:35:20 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2018/05/08 16:35:20 [debug] cassandra_keyspace = "kong"
2018/05/08 16:35:20 [debug] cassandra_lb_policy = "RoundRobin"
2018/05/08 16:35:20 [debug] cassandra_port = 9042
2018/05/08 16:35:20 [debug] cassandra_repl_factor = 1
2018/05/08 16:35:20 [debug] cassandra_repl_strategy = "SimpleStrategy"
2018/05/08 16:35:20 [debug] cassandra_schema_consensus_timeout = 10000
2018/05/08 16:35:20 [debug] cassandra_ssl = false
2018/05/08 16:35:20 [debug] cassandra_ssl_verify = false
2018/05/08 16:35:20 [debug] cassandra_timeout = 5000
2018/05/08 16:35:20 [debug] cassandra_username = "kong"
2018/05/08 16:35:20 [debug] client_body_buffer_size = "8k"
2018/05/08 16:35:20 [debug] client_max_body_size = "0"
2018/05/08 16:35:20 [debug] client_ssl = false
2018/05/08 16:35:20 [debug] custom_plugins = {"kong-authorization-plugin","kong-authentication-plugin","cerberus-plugin","kong-url-rewrite-plugin","kong-plugin-authorization","template-transformer"}
2018/05/08 16:35:20 [debug] database = "postgres"
2018/05/08 16:35:20 [debug] db_cache_ttl = 3600
2018/05/08 16:35:20 [debug] db_update_frequency = 5
2018/05/08 16:35:20 [debug] db_update_propagation = 0
2018/05/08 16:35:20 [debug] dns_error_ttl = 1
2018/05/08 16:35:20 [debug] dns_hostsfile = "/etc/hosts"
2018/05/08 16:35:20 [debug] dns_no_sync = false
2018/05/08 16:35:20 [debug] dns_not_found_ttl = 30
2018/05/08 16:35:20 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2018/05/08 16:35:20 [debug] dns_resolver = {}
2018/05/08 16:35:20 [debug] dns_stale_ttl = 4
2018/05/08 16:35:20 [debug] error_default_type = "text/plain"
2018/05/08 16:35:20 [debug] latency_tokens = true
2018/05/08 16:35:20 [debug] log_level = "notice"
2018/05/08 16:35:20 [debug] lua_package_cpath = ""
2018/05/08 16:35:20 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2018/05/08 16:35:20 [debug] lua_socket_pool_size = 30
2018/05/08 16:35:20 [debug] lua_ssl_verify_depth = 1
2018/05/08 16:35:20 [debug] mem_cache_size = "128m"
2018/05/08 16:35:20 [debug] nginx_daemon = "on"
2018/05/08 16:35:20 [debug] nginx_optimizations = true
2018/05/08 16:35:20 [debug] nginx_user = "nobody nobody"
2018/05/08 16:35:20 [debug] nginx_worker_processes = "auto"
2018/05/08 16:35:20 [debug] pg_database = "kong"
2018/05/08 16:35:20 [debug] pg_host = "127.0.0.1"
2018/05/08 16:35:20 [debug] pg_password = "******"
2018/05/08 16:35:20 [debug] pg_port = 5432
2018/05/08 16:35:20 [debug] pg_ssl = false
2018/05/08 16:35:20 [debug] pg_ssl_verify = false
2018/05/08 16:35:20 [debug] pg_user = "kong"
2018/05/08 16:35:20 [debug] prefix = "/usr/local/kong"
2018/05/08 16:35:20 [debug] proxy_access_log = "logs/access.log"
2018/05/08 16:35:20 [debug] proxy_error_log = "logs/error.log"
2018/05/08 16:35:20 [debug] proxy_listen = {"0.0.0.0:8000","0.0.0.0:8443 ssl"}
2018/05/08 16:35:20 [debug] real_ip_header = "X-Real-IP"
2018/05/08 16:35:20 [debug] real_ip_recursive = "off"
2018/05/08 16:35:20 [debug] server_tokens = true
2018/05/08 16:35:20 [debug] ssl_cipher_suite = "modern"
2018/05/08 16:35:20 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"
2018/05/08 16:35:20 [debug] trusted_ips = {}
2018/05/08 16:35:20 [debug] upstream_keepalive = 60
2018/05/08 16:35:20 [verbose] prefix in use: /usr/local/kong
2018/05/08 16:35:20 [verbose] preparing nginx prefix directory at /usr/local/kong
2018/05/08 16:35:20 [verbose] SSL enabled, no custom certificate set: using default certificate
2018/05/08 16:35:20 [verbose] default SSL certificate found at /usr/local/kong/ssl/kong-default.crt
2018/05/08 16:35:20 [verbose] Admin SSL enabled, no custom certificate set: using default certificate
2018/05/08 16:35:20 [verbose] admin SSL certificate found at /usr/local/kong/ssl/admin-kong-default.crt
2018/05/08 16:35:20 [warn] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
2018/05/08 16:35:20 [debug] searching for OpenResty 'nginx' executable
2018/05/08 16:35:20 [debug] /usr/local/openresty/nginx/sbin/nginx -v: 'nginx version: openresty/1.13.6.1'
2018/05/08 16:35:20 [debug] found OpenResty 'nginx' executable at /usr/local/openresty/nginx/sbin/nginx
2018/05/08 16:35:20 [debug] starting nginx: /usr/local/openresty/nginx/sbin/nginx -p /usr/local/kong -c nginx.conf
2018/05/08 16:35:20 [debug] nginx started
2018/05/08 16:35:20 [info] Kong started`

- Kong error logs: [error] 13997#0: *815304 SSL_do_handshake() failed (SSL: error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol) while SSL handshaking to upstream.
- Kong configuration: I have just this service registered.
- Operating System: Ubuntu 16.04

**Comment by thibaultcha** [comment]
Hi,

Thanks for the report, this has been fixed in 0.13.1 already, with #3358. See the [0.13.1 Changelog](https://github.com/Kong/kong/blob/master/CHANGELOG.md#0131---20180423) :)

**Comment by PFadel** [comment]
Oops, sorry about that.
